mongoengine/document.py

```diff
@@ -291,7 +288,12 @@ def save(self, force_insert=False, validate=True, clean=True,
 
         doc = self.to_mongo()
 
-        created = ('_id' not in doc or self._created or force_insert)
+        # I think the self._created flag is not necessarily required in PyMongo3
```

Comment:
Leave as it is and even more tests break. I'm not sure how to handle this, but PyMongo3 definitely changes the behaviour of save(), doing more frequently an update, which does not insert an object if it has already and strictly the primary key, even with upsert=True.

This is the point 2 of the comment in my PR and where I want feedback from many people. It is quite tricky and definitely a bigger conceptual change.